### PR TITLE
Bump bot-core to 8.2.1 and bump other deps to latest

### PR DESCRIPTION
**Pull request by ChrisLovering**
From the release notes on d.py [here](https://canary.discord.com/channels/336642139381301249/381965829857738772/1006359073610010664) and [here](https://canary.discord.com/channels/336642139381301249/381965829857738772/1008772795179737230) none of the breaking changes for the commit we were on to 2.0.1 affect us.

The biggest reason for this bump was to get onto d.py stable (better interactions support), and to use the new invite regex in bot core.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,52 +9,52 @@ license = "MIT"
...
+flake8 = "5.0.4"
+flake8-annotations = "2.9.1"
+flake8-bugbear = "22.9.23"
 flake8-docstrings = "1.6.0"
 flake8-string-format = "0.3.0"
 flake8-tidy-imports = "4.8.0"
 flake8-todo = "0.7"
-flake8-isort = "4.1.2.post0"
-pep8-naming = "0.13.1"
+flake8-isort = "5.0.0"
```
Comment (ChrisLovering) [inline pyproject.toml:49]: Breaking change in the major version bump was to remove support for Python 3.6.
https://pypi.org/project/flake8-isort/5.0.0/

File: pyproject.toml [hunk 2]
```diff
@@ -9,52 +9,52 @@ license = "MIT"
...
+pep8-naming = "0.13.2"
 pre-commit = "2.20.0"
-taskipy = "1.10.2"
 pip-licenses = "3.5.4"
-python-dotenv = "0.20.0"
-pytest = "7.1.2"
-pytest-cov = "3.0.0"
+pytest = "7.1.3"
+pytest-cov = "4.0.0"
```
Comment (ChrisLovering) [inline pyproject.toml:54]: Breaking change in the major version bump was to remove support multiprocessing, which we don't use.
https://pypi.org/project/pytest-cov/

File: pyproject.toml [hunk 3]
```diff
@@ -9,52 +9,52 @@ license = "MIT"
...
-tldextract = "3.3.1"
+tldextract = "3.4.0"
 
 [tool.poetry.dev-dependencies]
-coverage = "6.4.2"
-flake8 = "4.0.1"
-flake8-annotations = "2.9.0"
-flake8-bugbear = "22.7.1"
+coverage = "6.5.0"
+flake8 = "5.0.4"
```
Comment (ChrisLovering) [inline pyproject.toml:42]: Breaking changes in this major version bump don't affect us

https://flake8.pycqa.org/en/latest/release-notes/5.0.0.html

**Review by mbaruh — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 4]
```diff
@@ -9,52 +9,52 @@ license = "MIT"
...
-emoji = "2.0.0"
+emoji = "2.1.0"
 feedparser = "6.0.10"
-rapidfuzz = "2.3.0"
 lxml = "4.9.1"
 
 # Must be kept on this version unless doc command output is fixed
 # See https://github.com/python-discord/bot/pull/2156
-markdownify = "0.6.1"
+markdownify = "0.11.6"
```
[the remark this answers is not in the document]
Comment (ChrisLovering) [reply pyproject.toml]: No, and I completely ignored the commit directly above, which I left 😅.

Fixed in a forcepush.

**Review by mbaruh — APPROVED** [review]

**Review by ionite34 — APPROVED** [review]

**Pull request merged**